src/Compilers/CSharp/Portable/Symbols/Metadata/PE/MemberRefMetadataDecoder.cs

```diff
@@ -138,9 +138,8 @@ internal Symbol FindMember(MemberReferenceHandle memberRef, bool methodsOnly)
                             return null;
                         }
 
-                        ImmutableArray<ModifierInfo<TypeSymbol>> customModifiers;
-                        TypeSymbol type = this.DecodeFieldSignature(ref signaturePointer, out customModifiers);
-                        return FindFieldBySignature(_containingType, memberName, customModifiers, type);
+                        FieldInfo<TypeSymbol> fieldInfo = this.DecodeFieldSignature(ref signaturePointer);
+                        return FindFieldBySignature(_containingType, memberName, fieldInfo.CustomModifiers, fieldInfo.Type);
```

Comment:
>FindFieldBySignature [](http://example.com/codeflow?start=31&length=20)

It feels like ref-ness and ref-custom-modifiers should be taken into consideration

Reply:
I've added checks for `RefKind` and `RefCustomModifiers`, however it's not clear that we can hit a case where the decoded field does not match, even without the current `Type` or `CustomModifiers` comparisons.